### PR TITLE
Solution for #2

### PR DESCRIPTION
**Pull request by canndrew**

### CODE REVIEW

File: src/StratumClient/StratumClient.fs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+open System.IO
+open System.Text
+open Newtonsoft.Json
+
+type BlockchainTransactionGet = {
+    [<JsonProperty(PropertyName = "id")>]
+    mutable Id: int
+    [<JsonProperty(PropertyName = "result")>]
+    mutable Result: string
```
Comment (knocte) [inline src/StratumClient/StratumClient.fs]: are you sure you need the elements of the record to be mutable? I think there is no need

File: src/StratumClient/StratumClient.fs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        jsonWriter.WriteValue("blockchain.transaction.get")
+        jsonWriter.WritePropertyName("params")
+        jsonWriter.WriteStartArray()
+        jsonWriter.WriteValue(transactionId)
+        jsonWriter.WriteEnd()
+        jsonWriter.WriteEndObject()
+
+        let json = stringBuilder.ToString()
+        let! returnedJson = this.Call(json) |> Async.AwaitTask
+        let obj = JsonConvert.DeserializeObject<BlockchainTransactionGet>(returnedJson)
```
Comment (knocte) [inline src/StratumClient/StratumClient.fs]: cosmetic: no need to use parenthesis in the 2 lines above

File: src/Client/StratumClient.cs [hunk 3]
```diff
@@ -19,12 +20,23 @@ public class StratumClient : Client {
...
+            jsonWriter.WritePropertyName("id");
+            jsonWriter.WriteValue(id);
+            jsonWriter.WritePropertyName("method");
+            jsonWriter.WriteValue("blockchain.transaction.get");
+            jsonWriter.WritePropertyName("params");
+            jsonWriter.WriteStartArray();
+            jsonWriter.WriteValue(transactionId);
+            jsonWriter.WriteEnd();
+            jsonWriter.WriteEndObject();
+            var json = stringBuilder.ToString();
```
Comment (knocte) [inline src/Client/StratumClient.cs]: this is one way of using Newtonsoft.Json to write the JSON string, but maybe it's better that, in order to be consistent with how the deserialization is done, create a record type for the object to serialize to JSON? Example:

```
type Request =
    {
        Id: int;
        Method: string;
        Params: seq<obj>;
    }
```
(and adding attributes to map to their lowercase counterparts)

Comment (knocte) [reply src/Client/StratumClient.cs]: (then later use `JsonConvert.SerializeObject()`)

File: src/StratumClient/StratumClient.fs (hunk 1 above)
Comment (canndrew) [reply src/StratumClient/StratumClient.fs]: The C# class is public and has getters and setters. I don't know whether the setters are actually intended to be used anywhere, but I was trying to replicate the C# code faithfully.

File: src/Client/StratumClient.cs (hunk 3 above)
Comment (canndrew) [reply src/Client/StratumClient.cs]: Ah, this is what I was asking about in [my comment](https://github.com/nblockchain/TcpEchoSharp/pull/3/files#diff-95bf908641a93f7247c4e74ec3b4cd4aR30) :)

Though it was dumb of me to ask about it when I had an example already sitting there in the code.

**Comment by canndrew** [comment]
I've pushed a commit with the requested changes.

**Pull request merged**